### PR TITLE
Fixed: SentryHttpFailedRequestHandler disposes Content on .net framework

### PR DESCRIPTION
**Pull request by jamescrosswell**
Resolves https://github.com/getsentry/sentry-dotnet/issues/2684

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Fixes
+
+- Fixed: SentryHttpFailedRequestHandler disposes Content on .net framework ([#3306](https://github.com/getsentry/sentry-dotnet/pull/3306))
```
Comment (bitsandfoxes) [inline CHANGELOG.md:7]:
```suggestion
- The `SentryHttpFailedRequestHandler` now disposes content when targeting .NET framework ([#3306](https://github.com/getsentry/sentry-dotnet/pull/3306))
```

Comment (jamescrosswell) [reply CHANGELOG.md]: Updated the changelog... hopefully this is less ambiguous

File: src/Sentry/SentryHttpFailedRequestHandler.cs [hunk 2]
```diff
@@ -19,23 +20,17 @@ protected internal override void DoEnsureSuccessfulResponse([NotNull] HttpReques
...
-
         // Capture the event
         try
         {
+#if NET5_0_OR_GREATER
             response.EnsureSuccessStatusCode();
+#else
+            // Use our own implementation of EnsureSuccessStatusCode because older implementations of
+            // EnsureSuccessStatusCode disposes the content.
+            // See https://github.com/dotnet/runtime/issues/24845
```
Comment (bitsandfoxes) [inline src/Sentry/SentryHttpFailedRequestHandler.cs:31]: Appreciate those comments. 🙏

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Fixes
+
+-  `HttpResponse.Content` is no longer disposed by when using `SentryHttpFailedRequestHandler` on .NET Framework, which was causing an ObjectDisposedException when using Sentry with NSwag ([#3306](https://github.com/getsentry/sentry-dotnet/pull/3306))
```
Comment (jamescrosswell) [inline CHANGELOG.md:7]: @bitsandfoxes another attempt! Let me know if that's any better...

**Pull request merged**